ts/Extensions/RangeSelector.ts

```diff
@@ -2164,18 +2289,174 @@ class RangeSelector {
                     )
                 )
             ) {
-                // Move the input group down
-                inputGroup.attr({
-                    translateX: inputGroup.alignAttr.translateX + (
-                        chart.axisOffset[1] >= -xOffsetForExportButton ?
-                            0 :
-                            -xOffsetForExportButton
-                    ),
-                    translateY: inputGroup.alignAttr.translateY +
-                        buttonGroup.getBBox().height + 10
-                });
+                if (dropdown === 'responsive') {
+                    this.collapseButtons();
+                } else {
+                    // Move the input group down
+                    inputGroup.attr({
+                        translateX: inputGroup.alignAttr.translateX + (
+                            chart.axisOffset[1] >= -xOffsetForExportButton ?
+                                0 :
+                                -xOffsetForExportButton
+                        ),
+                        translateY: inputGroup.alignAttr.translateY +
+                            buttonGroup.getBBox().height + 10
+                    });
+                }
+            } else if (dropdown === 'responsive') {
+                this.showButtons();
+            }
+        }
+    }
 
+    /**
+     * Collapse the buttons and put the select element on top.
+     *
+     * @private
+     * @function Highcharts.RangeSelector#collapseButtons
+     * @return {void}
+     */
+    public collapseButtons(): void {
+        const {
+            buttons,
+            buttonOptions,
+            dropdown,
+            zoomText
+        } = this;
+
+        const getAttribs = (text: string): SVGAttributes => ({
+            text: `${text} ▾`,
+            paddingLeft: 8,
+            paddingRight: 8,
+            width: 'auto'
+        });
+
+        if (zoomText) {
+            zoomText.hide();
+        }
+
+        let hasActiveButton = false;
+
+        buttonOptions.forEach((
+            rangeOptions: Highcharts.RangeSelectorButtonsOptions,
+            i: number
+        ): void => {
+            const button = buttons[i];
+            if (button.state !== 2) {
+                button.hide();
+            } else {
+                button.show();
+                button.attr(getAttribs(rangeOptions.text));
+
+                hasActiveButton = true;
             }
+        });
+
+        if (!hasActiveButton && buttons.length > 0) {
+            if (dropdown) {
+                dropdown.selectedIndex = -1;
+            }
+
+            buttons[0].show();
+            buttons[0].attr(getAttribs(buttonOptions[0].text));
+        }
+
+        this.positionButtons();
+        this.showDropdown();
+    }
+
+    /**
+     * @private
+     * @function Highcharts.RangeSelector#showButtons
+     * @return {void}
+     */
+    public showButtons(): void {
```

Comment:
Is this the counterpart to `collapseButtons`? If so, maybe a better suited name is `expandButtons`?

Reply:
A little nitpicking here, but.. `showButtons` and `collapseButtons` are actually not only hiding/showing buttons, but also hide logic to show/hide the dropdown. Could we untangle this? What do you think?

Just in case, I assume `zoomText` as part of buttons and should be kept here

PS: Function description is missing in the doclet.

Reply:
I think they should stay tangled, they dont make much sense on their own and then you would have 2 function calls everywhere.